fxa-rust-client/src/scoped_keys.rs

```diff
@@ -0,0 +1,103 @@
+use errors::*;
+
+use base64;
+use byteorder::{ByteOrder, BigEndian};
+use ring;
+use ring::{aead, agreement, rand};
+use ring::agreement::EphemeralPrivateKey;
+use serde_json;
+use sha2::{Sha256, Digest};
+use untrusted::Input;
+
+pub struct ScopedKeysFlow {
+  private_key: EphemeralPrivateKey
+}
+
+impl ScopedKeysFlow {
+  pub fn with_random_ecdh_key() -> Result<ScopedKeysFlow> {
```

Comment:
nit: is the fact that it's "ecdh" important enough to expose in the name, or would something like "with_random_key" be sufficient?